lib/puppet/face/catalog.rb

```diff
@@ -33,6 +33,10 @@
     Puppet::Resource::Catalog object.
   EOT
 
+  unless find.options.include?(:facts_for_catalog)
```

Comment:
@joshcooper I feel like this is the best-bad option here, guarding here against trying to load the same `facts_for_catalog` option again in spec tests. However, there is another issue here with the `puppet catalog find` showing this option, with something like `--facts_for_catalog            - undocumented option`.  It seems like maybe we might just want to punt on that for the CLI and just make a note that that CLI option will just be ignored?

Reply:
I think you can set the summary directly:

```
    opt = Puppet::Interface::Option.new(find, '--facts_for_catalog')
    opt.summary = ""
    find.add_option(opt)
```